### PR TITLE
NONPR-1922 Update Trusts 5MLD schema to v1.2.0

### PR DESCRIPTION
**Pull request by crashposition**
Updates the schema and tests. Marked as WIP due to failing local tests. Raised PR for Craig to review and assist with testing.

### CODE REVIEW

**Comment by craigtd** [comment]
@crashposition I believe you will need to update https://github.com/hmrc/nonrep-generate-pdf/blob/master/src/main/resources/trusts-5mld.dito with a new template called `api-584-v1.2.0` (it will be identical to v1.1.0) for this to work.

**Comment by crashposition** [comment]
Regarding the workaround for templating. There is no non-breaking way to clone a template within a DITO project. We encountered this in December and it was discussed in this [Slack thread](https://itext.slack.com/archives/G01534HM5SN/p1609847549001400 ). The workaround was to reference the old template. See this [PR](https://github.com/hmrc/nonrep-generate-pdf/commit/907c8a6f3bb3386dc0fb8eb9cc442886fd2b53c6#diff-5c9b12ad0b1f0c82f0f65ed96173adabf61f1f38601423586090b5c195f32781).

File: src/it/resources/application.conf [hunk 1]
```diff
@@ -17,6 +17,14 @@ generate-pdf {
       pdf-template = "trusts-5mld.dito"
       signing-profile = "pades-t"
       api-key = "c93ea5ce61e3e92dc024536702a7375a5f4abc85556deae83375afcc6dead40f"
+    },
+    {
+      template-id = "trusts-5mld-1-2-0"
+      template-name = "api-584-v1.2.0"
```
Comment (craigtd) [inline src/it/resources/application.conf:23]: needs to be `api-584-v1.0.0`

File: src/test/resources/application.conf [hunk 2]
```diff
@@ -33,6 +33,14 @@ generate-pdf {
       pdf-template = "trusts-5mld.dito"
       signing-profile = "pades-t"
       api-key = "c93ea5ce61e3e92dc024536702a7375a5f4abc85556deae83375afcc6dead40f"
+    },
+    {
+      template-id = "trusts-5mld-1-2-0"
+      template-name = "api-584-v1.2.0"
```
Comment (craigtd) [inline src/test/resources/application.conf:39]: needs to be `api-584-v1.0.0`

Comment (crashposition) [reply src/test/resources/application.conf]: Change applied.

File: src/it/resources/application.conf (hunk 1 above)
Comment (crashposition) [reply src/it/resources/application.conf]: Change applied.

**Comment by crashposition** [comment]
@craigtd Have updated `trusts-5mld-1-2-0` in `application.conf` in `it` and `test` to match what we have in `main`. Getting three errors when I run tests:

```
[info] - should be successful on payload validation with correct schema v1.2.0 *** FAILED ***
[info]   false was not equal to true (ValidatorSpecs.scala:62)
[info]   org.scalatest.exceptions.TestFailedException:
```

```
[info] - should have template name accessible for pdf document template v1.2.0 *** FAILED ***
[info]   "api-584-v1.[0].0" was not equal to "api-584-v1.[2].0" (ServiceConfigSpec.scala:39)
```

```
[RoutesSpec-akka.actor.default-dispatcher-10] WARN a.a.t.ActorSystem - License usage event for PDF generation failure: expected json value got '{
        "...' (line 1, column 1)
[RoutesSpec-akka.actor.default-dispatcher-13] WARN a.a.t.ActorSystem - expected json value got '{
        "...' (line 1, column 1)
[info] - should accept request for generating pdf with valid v1.2.0 template *** FAILED ***
[info]   400 Bad Request was not equal to 200 OK (RoutesSpec.scala:119)
```

**Comment by crashposition** [comment]
WIP label removed. Test pass on Linux.

**Review by craigtd — APPROVED** [review]
Approving this PR on the caveat that all tests appear to pass on Linux but not on MacOS. Hopefully this will also work with our CodeBuild implementation allowing us to deliver the update. Can roll back if need be.

**Review by greg-gajda — APPROVED** [review]

**Comment by craigtd** [comment]
Tests failed via CodeBuild https://eu-west-2.console.aws.amazon.com/codesuite/codebuild/979211549557/projects/codebuild_k8s_services_generate-pdf_build_assembly/build/codebuild_k8s_services_generate-pdf_build_assembly%3A8884eff5-ca98-4156-b59c-ff3ff909b713/?region=eu-west-2

**Pull request merged**